src/v/coproc/tests/utils/wasm_event_generator.cc

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "coproc/tests/utils/wasm_event_generator.h"
+
+#include "bytes/bytes.h"
+#include "bytes/iobuf.h"
+#include "hashing/secure.h"
+#include "raft/types.h"
+#include "random/generators.h"
+#include "storage/record_batch_builder.h"
+
+model::record_header create_wasm_header(ss::sstring key, ss::sstring value) {
+    iobuf hkey, hval;
+    const auto key_size = key.size();
+    const auto val_size = value.size();
+    hkey.append(key.data(), key_size);
+    hval.append(value.data(), val_size);
+    return model::record_header(
+      key_size, std::move(hkey), val_size, std::move(hval));
+}
+
+model::record_header create_wasm_header(ss::sstring key, bytes value) {
+    iobuf hkey;
+    const auto key_size = key.size();
+    const auto val_size = value.size();
+    hkey.append(key.data(), key_size);
+    iobuf hval = bytes_to_iobuf(value);
+    return model::record_header(
+      key_size, std::move(hkey), val_size, std::move(hval));
+}
+
+/// Generates an event with randomized data fields, optionally bounded by event
+/// type
+model::record create_wasm_record(wasm_event e) {
+    iobuf key, value;
+    std::vector<model::record_header> headers;
+    if (e.uuid) {
+        key.append(e.uuid->data(), e.uuid->length());
+    }
+    if (e.script) {
+        value.append(e.script->data(), e.script->length());
+    }
+    if (e.desc) {
+        headers.emplace_back(
+          create_wasm_header("description", std::move(*e.desc)));
+    }
+    if (e.checksum) {
+        headers.emplace_back(
+          create_wasm_header("checksum", std::move(*e.checksum)));
+    }
+    if (e.action) {
+        ss::sstring action_value = (*e.action
+                                    == coproc::wasm_event_action::deploy)
+                                     ? "deploy"
+                                     : "remove";
+        headers.emplace_back(
+          create_wasm_header("action", std::move(action_value)));
+    }
+    /// TODO(rob) a bit of code duplication with
+    /// storage/test/utils/random_batch.cc
+    auto k_z = key.size_bytes();
+    auto v_z = value.size_bytes();
+    auto size = sizeof(model::record_attributes::type) // attributes
+                + vint::vint_size(0)                   // timestamp delta
```

Comment:
you should explain in comments that the reason deltas are 0 is because you only have 1 batch per.... but also why can't the batch_builder help you here?